client/src/App.js

```diff
@@ -632,6 +633,23 @@ class App extends Component {
                   <Redirect to={"/login"}/>
               }
           />
+          <Route exact path = {`/workspaces`} render={(props) =>
+                  this.state.loggedIn ?
```

Comment:
Ignorable nit, but 'twould be nice to have the login redirect logic in handled in one place before it gets to the router.